### PR TITLE
Fixing IDE warnings, fixing IEquatable, and simplifying code

### PR DESCRIPTION
**Pull request by palenshus**
* Updating code a little with more modern best practices (simpler IEquatable implementation, switch expressions, etc)
* Fixing IEquatable implemenation on ApiArray
* Fixing some IDE build warnings (unnecessary initialization, unused variables, etc)

### CODE REVIEW

File: src/WinGet.RestSource/WinGet.RestSource.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netstandard2.1</TargetFramework>
```
Comment (jamespik) [inline src/WinGet.RestSource/WinGet.RestSource.csproj:4]: To facilitate movement of common code to utilities libraries, netcoreapp3.1 makes it more straightforward. Is there any reason to keep this particular library netstandard?

**Review by jamespik — CHANGES_REQUESTED** [review]

File: src/WinGet.RestSource/WinGet.RestSource.csproj (hunk 1 above)
Comment (palenshus) [reply src/WinGet.RestSource/WinGet.RestSource.csproj:4]: Not that I'm aware of, I'm happy to go to .NET Core. Along those lines, any reason we shouldn't go straight to 5.0?

File: src/WinGet.RestSource.Functions/WinGet.RestSource.Functions.csproj [hunk 2]
```diff
@@ -49,6 +49,10 @@
         <None Update="host.json">
             <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
         </None>
+        <None Update="local.settings.template.json">
+          <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (zachcarp) [inline src/WinGet.RestSource.Functions/WinGet.RestSource.Functions.csproj:53]: >PreserveNewest [](http://example.com/codeflow?start=33&length=14)

Never

**Review by zachcarp — APPROVED** [review]
:shipit:

**Pull request merged**